tools/python/util/android/android.py

```diff
@@ -204,14 +214,126 @@ def start_emulator(
 
             _log.debug(f"sys.boot_completed='{getprop_value}'. Sleeping for {sleep_interval_seconds} before retrying.")
             time.sleep(sleep_interval_seconds)
-
+        # Verify if the emulator is now running
+        if not is_emulator_running_by_avd(avd_name=avd_name):
+            raise RuntimeError("Emulator failed to start.")
         return emulator_process
 
 
-def stop_emulator(emulator_proc_or_pid: typing.Union[subprocess.Popen, int]):
+def is_emulator_running_by_avd(avd_name: str) -> bool:
```

Comment:
Could we call these something like `check_emulator_running_using_avd_name`/`check_emulator_running_using_process`/`check_emulator_running_using_pid`?

The 'by' sounds like we could be checked how it was started rather than if it's currently running.